# Pharmaceutical lobbying

What is the extent of state-level lobbying by pharmaceutical interests?

## Import and process the data

In [1]:
import numpy as np
import pandas as pd

In [2]:
lobbying = pd.read_csv("data/ASAYLobClientsWCoding.txt", sep="\t", dtype={"ClientEID": object, "LobbyistEID": object, "Affiliate": object})
lobbying.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3165825 entries, 0 to 3165824
Data columns (total 10 columns):
Jurisdiction       object
Year               int64
ClientEID          object
Client             object
CatCodeGroup       object
CatCodeIndustry    object
CatCodeBusiness    object
LobbyistEID        object
Lobbyist           object
Affiliate          object
dtypes: int64(1), object(9)
memory usage: 241.5+ MB


In [3]:
lobbying.to_csv("data/nimp_lobbying.csv")

In [3]:
lobbying.head()

Jurisdiction  Year ClientEID                                    Client  \
0           NJ  2013  20317593                           GANNON, RICHARD   
1           NJ  2013  27607512                  180-TURNING LIVES AROUND   
2           NJ  2013  27420350                           EMMONS, WILLIAM   
3           NJ  2013  27535869                         FERNANDEZ, EDWARD   
4           NJ  2013  27606652  PALISADES SAFETY & INSURANCE ASSOCIATION   

  CatCodeGroup CatCodeIndustry CatCodeBusiness LobbyistEID         Lobbyist  \
0          NaN             NaN             NaN    20317593  GANNON, RICHARD   
1          NaN             NaN             NaN    20317593  GANNON, RICHARD   
2          NaN             NaN             NaN    27420350  EMMONS, WILLIAM   
3          NaN             NaN             NaN    27420350  EMMONS, WILLIAM   
4          NaN             NaN             NaN    27420350  EMMONS, WILLIAM   

  Affiliate  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [4]:
lobbying.groupby("Year")["Jurisdiction"].nunique()

Year
2000     2
2001     2
2002    10
2003    12
2004    12
2005    17
2006    50
2007    50
2008    50
2009    50
2010    50
2011    50
2012    50
2013    50
2014    50
2015    50
2016    50
2017    34
Name: Jurisdiction, dtype: int64

Filter to just 2006-2016.

In [5]:
lobbying = lobbying[(lobbying["Year"] >= 2006) & (lobbying["Year"] <= 2016)]
lobbying.reset_index(drop=True, inplace=True)
lobbying.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2713150 entries, 0 to 2713149
Data columns (total 10 columns):
Jurisdiction       object
Year               int64
ClientEID          object
Client             object
CatCodeGroup       object
CatCodeIndustry    object
CatCodeBusiness    object
LobbyistEID        object
Lobbyist           object
Affiliate          object
dtypes: int64(1), object(9)
memory usage: 207.0+ MB


In [6]:
lobbying.head()

Jurisdiction  Year ClientEID                                    Client  \
0           NJ  2013  20317593                           GANNON, RICHARD   
1           NJ  2013  27607512                  180-TURNING LIVES AROUND   
2           NJ  2013  27420350                           EMMONS, WILLIAM   
3           NJ  2013  27535869                         FERNANDEZ, EDWARD   
4           NJ  2013  27606652  PALISADES SAFETY & INSURANCE ASSOCIATION   

  CatCodeGroup CatCodeIndustry CatCodeBusiness LobbyistEID         Lobbyist  \
0          NaN             NaN             NaN    20317593  GANNON, RICHARD   
1          NaN             NaN             NaN    20317593  GANNON, RICHARD   
2          NaN             NaN             NaN    27420350  EMMONS, WILLIAM   
3          NaN             NaN             NaN    27420350  EMMONS, WILLIAM   
4          NaN             NaN             NaN    27420350  EMMONS, WILLIAM   

  Affiliate  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

How many records contain "Uncoded" category group code information and what are the category business codes under which they fall?

In [7]:
lobbying[lobbying["CatCodeGroup"] == "Uncoded"].groupby("CatCodeBusiness").size()

CatCodeBusiness
Employer listed but category unknown                   687
Generic occupation - impossible to assign category      66
Homemakers, students & other non-income earners        108
No employer listed or discovered                      1295
Uncoded                                                654
dtype: int64

And how many contain "NaN" values?

In [8]:
lobbying[lobbying["CatCodeGroup"].isnull()]

Jurisdiction  Year ClientEID  \
0                 NJ  2013  20317593   
1                 NJ  2013  27607512   
2                 NJ  2013  27420350   
3                 NJ  2013  27535869   
4                 NJ  2013  27606652   
5                 NJ  2013  27420350   
6                 NJ  2013  27535869   
7                 NJ  2013  27606652   
8                 NJ  2013  28229120   
10                NJ  2013  28229121   
11                NJ  2013  31822595   
12                NJ  2013  27452018   
13                NJ  2013  27535856   
15                NJ  2013  27535938   
16                NJ  2013  27535812   
17                NJ  2013  27498322   
18                NJ  2013  27557555   
19                NJ  2013  28229122   
21                NJ  2013  27452018   
22                NJ  2013  27535856   
24                NJ  2013  27535938   
25                NJ  2013  27535812   
26                NJ  2013  27498322   
27                NJ  2013  27557555   
28                NJ  2013  28229122   
30                NJ  2013  27452018   
31                NJ  2013  27535856   
33                NJ  2013  27535938   
34                NJ  2013  27535812   
35                NJ  2013  27498322   
...              ...   ...       ...   
2713094           IL  2013  27846779   
2713095           IL  2013  27684038   
2713097           IL  2013  27597195   
2713098           IL  2013  27684038   
2713100           IL  2013  27846794   
2713102           IL  2013  27684055   
2713103           IL  2013  27670210   
2713104           IL  2013  27674634   
2713105           IL  2013  27835063   
2713107           IL  2013  27684055   
2713108           IL  2013  27670210   
2713109           IL  2013  27674634   
2713110           IL  2013  27846795   
2713112           IL  2013  27597150   
2713113           IL  2013  27597150   
2713114           IL  2013  27846808   
2713115           IL  2013  27684082   
2713116           IL  2013  27835642   
2713120           IL  2013  27597581   
2713122           IL  2013  27559194   
2713124           IL  2013  27483629   
2713126           IL  2013  27551451   
2713130           IL  2013  27598431   
2713132           IL  2013  27684284   
2713137           IL  2013  27684284   
2713142           ND  2013  27598603   
2713144           ND  2013  27565436   
2713146           SD  2013  27598986   
2713147           SD  2013  27710532   
2713148           SD  2013  27532970   

                                           Client CatCodeGroup  \
0                                 GANNON, RICHARD          NaN   
1                        180-TURNING LIVES AROUND          NaN   
2                                 EMMONS, WILLIAM          NaN   
3                               FERNANDEZ, EDWARD          NaN   
4        PALISADES SAFETY & INSURANCE ASSOCIATION          NaN   
5                                 EMMONS, WILLIAM          NaN   
6                               FERNANDEZ, EDWARD          NaN   
7        PALISADES SAFETY & INSURANCE ASSOCIATION          NaN   
8                                    HART, DENNIS          NaN   
10                              FRIEDLANDER, EZRA          NaN   
11                           XCHANGE TELECOM CORP          NaN   
12                                RYAN, ELIZABETH          NaN   
13                             EDELSTEIN, THERESA          NaN   
15                                  HOPKINS, SEAN          NaN   
16                                 COHEN, JESSICA          NaN   
17                                   EICHER, NEIL          NaN   
18                                 LECHNER, SARAH          NaN   
19                                  HOUSMAN, TYLA          NaN   
21                                RYAN, ELIZABETH          NaN   
22                             EDELSTEIN, THERESA          NaN   
24                                  HOPKINS, SEAN          NaN   
25                                 COHEN, JESSICA          NaN   
26       

OK. 2,810 records have an "Uncoded" category group code and another 1,579,636 records have an "NaN" value. Those figures represent a combined 58.3 percent of the 2,713,150 records from 2006 though 2016.

## Analysis

Where does the pharmaceutical industry rank in terms of how many lobbyists they've hired?

We're interested in lobbyist registrations by year: that is, how many contracts exist between pharmaceutial clients and lobbyists in each state and within each year? Because some states require lobbyists to register multiple times a year, we need to group by year, state, client and lobbyist in order to avoid duplicates.

In [49]:
registrations = lobbying.groupby(["Year", "Jurisdiction", "CatCodeBusiness", "Client", "Lobbyist"], as_index=False)
registrations.to_frame()

AttributeError: 'DataFrameGroupBy' object has no attribute 'to_frame'

In [31]:
cleaned = lobbying.astype(str).groupby(["Year", "Jurisdiction", "CatCodeBusiness", "ClientEID", "LobbyistEID"]).sum()
cleaned = cleaned.reset_index()
cleaned_grouped_by_business_code = cleaned.groupby("CatCodeBusiness").sum()
cleaned_grouped_by_business_code = cleaned_grouped_by_business_code.reset_index(name="relationships")
cleaned_grouped_by_business_code.sort_values("relationships", ascending=False)

KeyboardInterrupt: 

In [40]:
lobbying[lobbying["CatCodeBusiness"].isnull()]

Jurisdiction  Year ClientEID  \
0                 NJ  2013  20317593   
1                 NJ  2013  27607512   
2                 NJ  2013  27420350   
3                 NJ  2013  27535869   
4                 NJ  2013  27606652   
5                 NJ  2013  27420350   
6                 NJ  2013  27535869   
7                 NJ  2013  27606652   
8                 NJ  2013  28229120   
10                NJ  2013  28229121   
11                NJ  2013  31822595   
12                NJ  2013  27452018   
13                NJ  2013  27535856   
15                NJ  2013  27535938   
16                NJ  2013  27535812   
17                NJ  2013  27498322   
18                NJ  2013  27557555   
19                NJ  2013  28229122   
21                NJ  2013  27452018   
22                NJ  2013  27535856   
24                NJ  2013  27535938   
25                NJ  2013  27535812   
26                NJ  2013  27498322   
27                NJ  2013  27557555   
28                NJ  2013  28229122   
30                NJ  2013  27452018   
31                NJ  2013  27535856   
33                NJ  2013  27535938   
34                NJ  2013  27535812   
35                NJ  2013  27498322   
36                NJ  2013  27557555   
37                NJ  2013  28229122   
39                NJ  2013  27452018   
40                NJ  2013  27535856   
42                NJ  2013  27535938   
43                NJ  2013  27535812   
44                NJ  2013  27498322   
45                NJ  2013  27557555   
46                NJ  2013  28229122   
48                NJ  2013  27452018   
49                NJ  2013  27535856   
51                NJ  2013  27535938   
52                NJ  2013  27535812   
53                NJ  2013  27498322   
54                NJ  2013  27557555   
55                NJ  2013  28229122   
57                NJ  2013  27452018   
58                NJ  2013  27535856   
60                NJ  2013  27535938   
61                NJ  2013  27535812   
62                NJ  2013  27498322   
63                NJ  2013  27557555   
64                NJ  2013  28229122   
66                NJ  2013  27452018   
67                NJ  2013  27535856   
69                NJ  2013  27535938   
70                NJ  2013  27535812   
71                NJ  2013  27498322   
72                NJ  2013  27557555   
73                NJ  2013  28229122   
75                NJ  2013  27452018   
76                NJ  2013  27535856   
78                NJ  2013  27535938   
79                NJ  2013  27535812   
80                NJ  2013  27498322   
81                NJ  2013  27557555   
82                NJ  2013  28229122   
84                NJ  2013  27451823   
85                NJ  2013  27451685   
86                NJ  2013  27451794   
87                NJ  2013  27420615   
88                NJ  2013  27420744   
89                NJ  2013  16394761   
90                NJ  2013  27498462   
91                NJ  2013  16394760   
92                NJ  2013  27535920   
93                NJ  2013  27535776   
94                NJ  2013  27535898   
95                NJ  2013  27557385   
96                NJ  2013  28229123   
97                NJ  2013  28229124   
98                NJ  2013  28229125   
100               NJ  2013  27451823   
101               NJ  2013  27451685   
102               NJ  2013  27451794   
103               NJ  2013  27420615   
104               NJ  2013  27420744   
105               NJ  2013  16394761   
106               NJ  2013  27498462   
107               NJ  2013  16394760   
108               NJ  2013  27535920   
109               NJ  2013  27535776   
110               NJ  2013  27535898   
111               NJ  2013  27557385   
112               NJ  2013  28229123   
113               NJ  2013  28229124   
114               NJ  2013  28229125   
116               NJ  2013  27451823   
117               NJ  2013  27451685   
118     

In [36]:
business_categories = lobbying.groupby("CatCodeBusiness").size()
business_categories = business_categories.reset_index(name="records")
business_categories
business_categories["rank"] = business_categories["records"].rank(method="min", ascending=False).astype(int)
business_categories.sort_values("records", ascending=False)

CatCodeBusiness  records
0                             AIDS treatment & testing      185
1                          Abortion policy, pro-choice      694
2                            Abortion policy, pro-life      504
3                          Accident & health insurance    18927
4                                          Accountants     8088
5    Actors, actresses & others in the live theater...     2363
6                                 Adhesives & sealants      115
7              Advertising & public relations services     3017
8    Agricultural chemicals (fertilizers & pesticides)     1588
9                            Agricultural labor unions       35
10          Agricultural services & related industries     1229
11                                         Air freight        3
12                                       Air transport      834
13                                Air transport unions       36
14                              Aircraft manufacturers     1692
15                                            Airlines     2231
16                                             Alcohol     1095
17            Alternative energy production & services     7844
18                        Aluminum mining & processing      173
19                      Amusement & recreation centers     1104
20                                     Amusement parks      434
21                       Animal feed & health products      197
22                                       Animal rights     2272
23                                    Anti-gun control     2402
24                          Apparel & accessory stores       20
25                              Architectural services     2104
26              Artificial sweeteners & food additives       33
27                               Attorneys & law firms    42099
28                            Auto dealers, new & used     5940
29                                  Auto manufacturers     6443
30                                         Auto repair      814
31                               Automotive & trucking      380
32                                   Automotive unions      225
33                        Aviation services & airports      505
34                                  Bail Bond Services     1050
35                        Banks & lending institutions     1246
36                               Beauty & barber shops      918
37                                                Beer     4307
38                           Beverages (non-alcoholic)     2728
39   Bicycles & other non-motorized recreational tr...      152
40                         Biotech products & research     3674
41             Book, newspaper & periodical publishing     4401
42                      Broadcasting & motion pictures     1509
43                               Builders associations     7779
44                                  Building materials      826
45                       Building operators & managers     4135
46                                        Bus services     1455
47                                       Buses & taxis      192
48                                   Business services     2029
49                Campaign funding from public sources        1
50                              Candidate Contribution     4521
51                                Candidate committees      151
52                         Car & truck rental agencies     3350
53                      Casinos, racetracks & gambling    19834
54                         Catalog & mail order houses       33
55                                Chambers of commerce     7112
56                                           Chemicals     4784
57                                   Children's rights     1452
58                                       Chiropractors     1389
59                Christian Coalition, religious right     1200
60                       Civil servant/public employee     5673
61                   Civil service & government unions      199
62     

In terms of the number of lobbyist-client relationships, pharmaceutical manufacturing ranks third with 37,687 records and medical supplies manufacturing & sales ranks 77th with 4,199 records.

How many clients from the pharmaceutical manufacturing or medical supplies manufacturing & sales business categories hired lobbyists between 2006 and 2016?

In [22]:
medical_companies = lobbying[(lobbying["CatCodeBusiness"] == "Pharmaceutical manufacturing") | (lobbying["CatCodeBusiness"] == "Medical supplies manufacturing & sales")].groupby("Client").size()
medical_companies = medical_companies.reset_index(name="records")
medical_companies["rank"] = medical_companies["records"].rank(method="min", ascending=False).astype(int)
medical_companies.sort_values("records", ascending=False).head()

Client  records  rank
289  PHARMACEUTICAL RESEARCH & MANUFACTURERS ASSOCI...     2917     1
287                                             PFIZER     2770     2
26                                         ASTRAZENECA     1682     3
316                                     SANOFI-AVENTIS     1467     4
112                                     ELI LILLY & CO     1440     5

In [23]:
medical_companies["Client"].nunique()

385

OK. So, it looks like 385 clients from the pharmaceutical manufacturing or medical supplies manufacturing & sales business categories hired lobbyists between 2006 and 2016.

How many lobbyists did these clients hire between 2006 and 2016?

In [24]:
medical_lobbyists = lobbying[(lobbying["CatCodeBusiness"] == "Pharmaceutical manufacturing") | (lobbying["CatCodeBusiness"] == "Medical supplies manufacturing & sales")].groupby("Lobbyist").size()
medical_lobbyists = medical_lobbyists.reset_index(name="records")
medical_lobbyists["rank"] = medical_lobbyists["records"].rank(method="min", ascending=False).astype(int)
medical_lobbyists.sort_values("records", ascending=False).head()

Lobbyist  records  rank
5571  SETZEPFANDT, SCOTT      254     1
2100   GALLO, GEOFFREY A      149     2
1153    COFFEE, SHERRI D      124     3
6360      VOJTECH, JULIE      120     4
6256       TURNER, JAMES      113     5

In [25]:
medical_lobbyists["Lobbyist"].nunique()

6776

OK. So, it looks like these clients hired 6,776 lobbyists between 2006 and 2016.

How do the number of lobbyists vary from year to year between 2006 and 2016?

In [28]:
lobbyists_by_year = lobbying[(lobbying["CatCodeBusiness"] == "Pharmaceutical manufacturing") | (lobbying["CatCodeBusiness"] == "Medical supplies manufacturing & sales")].groupby("Year")["Lobbyist"].nunique()
lobbyists_by_year = lobbyists_by_year.reset_index(name="lobbyists")
lobbyists_by_year["rank"] = lobbyists_by_year["lobbyists"].rank(method="min", ascending=False).astype(int)
lobbyists_by_year.sort_values("Year", ascending=False)

Year  lobbyists  rank
10  2016       2152     2
9   2015       2269     1
8   2014       2064     3
7   2013       2061     4
6   2012       1692    11
5   2011       2048     5
4   2010       1897     8
3   2009       1747    10
2   2008       1935     7
1   2007       1812     9
0   2006       1952     6

In [ ]:
state, year, client, lobbyist = individual registrations
state, year, lobbyist = individual lobbyists